cmd/check.go

```diff
@@ -220,7 +221,39 @@ func check() {
 				if err != nil {
 					log.Fatalf("Cannot create k8s client: %+v", err)
 				}
-				dg = k8s.NewPodsDataGatherer(k8sClient)
+				dg = k8s.NewGenericGatherer(k8sClient, schema.GroupVersionResource{
+					Group:    "",
+					Version:  "v1",
+					Resource: "pods",
+				})
+			} else if strings.HasPrefix(name, "k8s/") {
+				trimmed := strings.TrimPrefix(name, "k8s/")
+				nameOnDots := strings.SplitN(trimmed, ".", 3)
+				// if a user has used for example `k8s/pods.v1`, we should
+				// handle this case and set the group name to empty.
+				if len(nameOnDots) == 2 {
+					nameOnDots = append(nameOnDots, "")
+				}
+				if len(nameOnDots) != 3 {
+					log.Fatal("Failed to parse generic k8s plugin configuration. Expected data gatherer name of the form k8s/{resource-name}.{api-version}.{api-group}")
+				}
```

Comment:
Doesn't this break compatibility with existing configurations using `k8s/pods`?

Reply:
I think it should be possible to keep this with a manual case for `k8s/pod` -> `k8s/pods.v1` as the output will still be grouped under the datagatherer name (https://github.com/jetstack/preflight/pull/99/files#diff-f9c4c955b1d6c71769486619f542fef1R267)

Reply:
yes, but that "mapping" is needed here.

Reply:
No, I haven't removed the old code - `k8s/pod` is the block just above this.

Reply:
https://github.com/jetstack/preflight/blob/4423174dc2013aa09b1434296999291fc3119124/cmd/check.go#L211-L228

Reply:
ups, I missed that.